build/run-in-docker

```diff
@@ -60,16 +60,45 @@ else
   RUN_CMD="${@@Q}"
 fi
 
+RO_MOUNTS=(
+  "/etc/group"
+  "/etc/passwd"
+  "/etc/shadow"
+  "/etc/sudoers.d"
+)
+
+RW_MOUNTS=(
+  "$GIT_COMMON_DIR"
+  "$WORKDIR"
+  "$LOCAL_CCACHE_DIR"
+  "$LOCAL_MAVEN_REPO"
+)
+
+function mount_args() {
+  local mount_mode=$1
+  shift
+  local mounts=("$@")
+  for m in "${mounts[@]}"
+  do
+    if [[ "$mount_mode" == "ro" ]]
+    then
+      printf -- ' --mount type=bind,src=%s,dst=%s,%s\n' "$m" "$m" "$mount_mode"
+    else
+      # rw is default
+      printf -- ' --mount type=bind,src=%s,dst=%s\n' "$m" "$m"
+    fi
+  done
+}
+
+if [[ "$HOST_CUDA_BIN_PATH" != "" ]]
```

Comment:
I think the `BIN_PATH` is not accurate, as we need to specify the entire CTK path including `bin` path and `lib` and `lib64` paths etc. So saying `BIN` is misleading.
```suggestion
if [[ "$HOST_CUDA_ROOT_PATH" != "" ]]
```

Reply:
Then how about `HOST_CUDA_TOOLKIT_ROOT_DIR`? The suffix `_ROOT_DIR` is more accurate than `_BIN_PATH`.

Reply:
I did a quick search and found this:
```
Make sure that after installation that you have CUDA_BIN_PATH, CUDA_INC_PATH and CUDA_LIB_PATH set to the location of where CUDA was installed (e.g., C:\CUDA\bin, C:\CUDA\include, C:\CUDA\lib) 
```
So indeed, `CUDA_BIN_PATH` should point to the `CUDA_TOOLKIT_ROOT_PATH/bin` directory. The CMake doc seems incorrect.

Reply:
Oh wow, also found another env variable: `CUDA_PATH`, which is shorter and seems to be better than `CUDA_TOOLKIT_ROOT_DIR`: https://cmake.org/cmake/help/latest/module/FindCUDAToolkit.html.

So how about `HOST_CUDA_PATH`?

Reply:
👍 FWIW, I liked the idea of `HOST_CUDA_TOOLKIT_ROOT_DIR`, personally.  It is explicit about being the "root".

But I'm happy to go with what you two work out.